plugins/nexus-repository-manager/src/components/NexusRepositoryManager/NexusRepositoryManager.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { useAsync } from 'react-use';
+
+import { Entity } from '@backstage/catalog-model';
+import { Link, Progress, Table } from '@backstage/core-components';
+
+import { Box, Chip, makeStyles } from '@material-ui/core';
+
+import { formatByteSize, formatDate } from '@janus-idp/shared-react';
+
+import { type NexusRepositoryManagerApiV1 } from '../../classes';
+import { useNexusRepositoryManagerAppData } from '../../hooks';
+import { Annotation } from '../../types';
+import { getFileSize, getSha256 } from '../../utils';
+import { columns, useStyles } from './tableHeading';
+
+const useLocalStyles = makeStyles({
+  chip: {
+    margin: 0,
+    marginRight: '.2em',
+    height: '1.5em',
+    '& > span': {
+      padding: '.3em',
+    },
+  },
+});
+
+type NexusRepositoryManagerProps = {
+  nexusClient: NexusRepositoryManagerApiV1;
+  entity: Entity;
+  ANNOTATIONS: Readonly<Annotation[]>;
+};
+
+export function NexusRepositoryManager(props: NexusRepositoryManagerProps) {
```

Comment:
```suggestion
export const NexusRepositoryManager(props: NexusRepositoryManagerProps) => {
```

Reply:
Stylistically I like to use the function keyword whenever its not a callback. Moreover, I would suggest reading [this](https://www.reddit.com/r/reactjs/comments/s9u4eq/react_arrow_function_component_or_react_function/hty1u3u/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) as to why I prefer the function keyword.

> "Function" functions wherever possible. The only time arrow functions are justifiable in general, not just for components, are one-off/quick/small definitions IMO.

> - "Function" results in more readable/organized code. Seeing "function" makes the component stick out more than arrows, esp when those are mixed with other arrows.
> - Besides avoiding mixing with other arrows, if I see a stream of const/let/var declarations in a row, it's easy for me to gloss over them and then when the arrow function is called, I'm confused for a short time, breaking the principle of least astonishment.
> - Usually, components are longer than one line of code, so you don't even get the benefit of the auto-return property of arrows.

> While I'm obv quite biased in this respect, some more objective points brought up by others:

> - It's the default behavior for the [react/function-component-definition](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md) ESLint rule, which suggests to me that it's what the React devs recommend.

> - You can define them in the same line as export default.

> - It's easier with TypeScript (FC, and to a lesser extent, FunctionComponent are [best avoided](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/), TS will infer the type automatically).

> - They automatically have a (display)Name injected, particularly when wrapped with another function like forwardRefor memo (last I checked, arrows don't but that could've been fixed in a minor/patch update).

> - They're hoisted, so you can put it and other functions in any order to improve readability of the whole file without "x isn't defined" errors (e.g. most important functions/components at the top of the file and minor/utility functions at the bottom, or vice versa).